src/display/annotation_layer.js

```diff
@@ -668,7 +668,7 @@ class TextWidgetAnnotationElement extends WidgetAnnotationElement {
         if (elementData.formattedValue) {
           event.target.value = elementData.formattedValue;
         }
-        event.target.setSelectionRange(0, 0);
+        event.target.scrollLeft = 0;
```

Comment:
This *really* needs a good comment, referencing the original issue, since it's less clear exactly what this does than in the previous code; I'd suggest: 
```javascsript
        // Reset the cursor position to the start of the field (issue 12359).
        event.target.scrollLeft = 0;
```

When fixing this, please remember https://github.com/mozilla/pdf.js/wiki/Squashing-Commits